### PR TITLE
Adapt GetMetricHistory and GetMetricHistories to return context

### LINKED ISSUE #207: Metric contexts: extend the MLFlow `GetMetricHistory` and `GetMetricHistories` endpoints to return an extra field `context` for each metric
- Add to the MLFLow API endpoints
- return the full context in the response for all requests
- test MLFLow client against `GetMetricHistory` to ensure it does not break with this change
- `GetMetricHistoryBulk` does _not_ need to be changed

### PR DESCRIPTION
**Pull request by fabiovincenzi**
fixes #207 
Adapt `GetMetricHistory` and `GetMetricHistories` to work with context.

### CODE REVIEW

File: pkg/api/mlflow/api/response/metric.go [hunk 1]
```diff
@@ -1,57 +1,78 @@
...
 }
 
+// MetricPartialResponse is a partial response object for GetMetricHistoryResponse.
+type MetricPartialResponse struct {
+	RunID     string         `json:"run_id,omitempty"`
+	Key       string         `json:"key"`
+	Value     any            `json:"value"`
+	Timestamp int64          `json:"timestamp"`
+	Step      int64          `json:"step"`
+	Context   map[string]any `json:"context,omitempty"`
```
Comment (dsuhinin) [inline pkg/api/mlflow/api/response/metric.go:28]: not sure, but `omitempty` is is something really required?

**Review by dsuhinin — APPROVED** [review]
Looks good!

File: pkg/api/mlflow/api/response/metric.go [hunk 2]
```diff
@@ -1,57 +1,78 @@
...
 
 // NewMetricHistoryResponse creates new GetMetricHistoryResponse object.
-func NewMetricHistoryResponse(metrics []models.Metric) *GetMetricHistoryResponse {
+func NewMetricHistoryResponse(metrics []models.Metric) (*GetMetricHistoryResponse, error) {
 	resp := GetMetricHistoryResponse{
 		Metrics: make([]MetricPartialResponse, len(metrics)),
 	}
 
 	for n, m := range metrics {
+		var context map[string]interface{}
```
Comment (dsuhinin) [inline pkg/api/mlflow/api/response/metric.go:43]: let initialise it inside the:
```
if m.Context != nil
```

File: pkg/api/mlflow/dao/repositories/metric.go [hunk 3]
```diff
@@ -198,6 +198,8 @@ func (r MetricRepository) GetMetricHistories(
 	).Joins(
 		"INNER JOIN experiments ON experiments.experiment_id = runs.experiment_id AND experiments.namespace_id = ?",
 		namespaceID,
+	).Joins(
+		"Context",
```
[the remark this answers is not in the document]
Comment (dsuhinin) [reply pkg/api/mlflow/dao/repositories/metric.go:202]: If I remember correctly is was only one way to `auto` populate `Context` object.

[the remark this answers is not in the document]
Comment (suprjinx) [reply pkg/api/mlflow/dao/repositories/metric.go:202]: image

this is left join

**Review by suprjinx — APPROVED** [review]

**Pull request merged**